CodeU_Assignement_1/src/codeu_assignement_1/SinglyLinkedList.java

```diff
@@ -0,0 +1,236 @@
+        {
+            next = null;
+            data = null;
+        }    
+        /*  C'tor  */
+        public Node(T d,Node n)
+        {
+            data = d;
+            next = n;
+        }    
+        // Setters
+        public void setNext(Node n) { next = n; }    
+        public void setData(T d) { data = d; } 
+        
+        // Getter
+        public Node getNext() { return next; }    
+        public T getData() { return data; }
+    }
+    
+    
+    /* Data members of LinkedList */
+    protected int size=0;
+    protected Node m_first=null;
+    protected Node m_last=null;
+    
+    // Default c'tor
+    public SinglyLinkedList(){} 
+
+    // C'tor: adds all items of another collection
+    public SinglyLinkedList(Collection<? extends E> c) { this.addAll(c); } 
+            
+    /*  Check if the list is empty  */
+    public boolean isEmpty() { return (m_first == null); }
+
+    /*  Get the current size of the list  */
+    public int getSize() { return size; }
+ 
+    /*  Add e element in the front/begining of the list  */
+    public void addFirst(E e) { addAt(0,e); }
+    
+    /*  Add e element in the tail/end of the list  */
+    public void addLast(E e) { addAt(size,e); }
+    
+    /*  Add e element at index  */
+    public void addAt(int index, E e)
+    {        
+        // check - index is begger than size or neg.
+        if (index < 0 || index > size)
+            throw new IndexOutOfBoundsException();
+     
+        Node n = new Node(e, null);             
+                
+        // Empty list
+        if(m_first == null) 
+        {
+            m_first = n;
+            m_last = m_first;
+        }
+        // Add to first
+        else if (index == 0)
+        {
+            n.setNext(m_first);
+            m_first = n;
+        }
+        // Add to last
+        else if (index == size)
+        {
+            m_last.setNext(n);
+            m_last = n;
+        }
+        // Add to in the middle
+        else 
+        {
+            Node curr = m_first;
+            index -= 1; // i = i-1; to get the item befoer where we add
+            for (int i = 0; i < size; i++) 
+            {
+                if (i == index) 
+                {
+                    // Before: curr -> next_curr
+                    Node next_curr = curr.getNext() ;
+                    curr.setNext(n);
+                    n.setNext(next_curr);
+                    break;
+                    // After: curr -> n -> next_curr
+                }
+
+                // Go to next
+                curr = curr.getNext();
+            }
+        }
+        
+        size++ ;
+    }
+    
+    /*
+        Add all the items in the collection into the list
+        NullPointerException - if the specified collection is null
+    */
+    public boolean addAll(Collection<? extends E> c)
+    {
+        if (c == null)
+            throw new NullPointerException();
+        
+        Iterator <? extends E> itr = c.iterator();

```

Comment:
It's the same in terms of the code generated.

It's different in terms of ease of writing of the code, the readability of the code, and the maintainability of the code. There is a reason it was added to the language.